voyager-lifecycle-kmp/src/androidMain/kotlin/cafe/adriel/voyager/jetpack/AndroidScreenLifecycleOwner.android.kt

```diff
@@ -0,0 +1,59 @@
+package cafe.adriel.voyager.jetpack
+
+import android.app.Activity
+import android.app.Application
+import android.content.Context
+import android.content.ContextWrapper
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ProvidedValue
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalSavedStateRegistryOwner
+import androidx.lifecycle.SavedStateViewModelFactory
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewmodel.MutableCreationExtras
+import androidx.savedstate.SavedStateRegistryOwner
+import java.util.concurrent.atomic.AtomicReference
+
+internal actual class SavedStateViewModelPlatform actual constructor(val owner: SavedStateRegistryOwner) {
+    private val atomicContext = AtomicReference<Context>()
+
+    actual fun getDefaultViewModelProviderFactory(): ViewModelProvider.Factory =
+        SavedStateViewModelFactory(
+            application = atomicContext.get()?.applicationContext?.getApplication(),
+            owner = owner
+        )
+
+    actual fun providePlatform(extras: MutableCreationExtras) {
+        val application = atomicContext.get()?.applicationContext?.getApplication()
+        if (application != null) {
+            extras.set(ViewModelProvider.AndroidViewModelFactory.APPLICATION_KEY, application)
+        }
```

Comment:
What would happen if we didn't set the extras in case the application is null? Maybe you could log it as a warning?